### PR TITLE
JSON schema

### PR DESCRIPTION
**Pull request by RayMathew**
1. Added schema that JSON files must adhere to.
2. Fixed all JSON files that did not adhere to the schema.

### CODE REVIEW

**Comment by RayMathew** [comment]
Changed to draft, found scope for further improvement.

**Comment by RayMathew** [comment]
Readt for review

File: src/content/config.ts [hunk 1]
```diff
@@ -2,7 +2,18 @@ import { defineCollection, z } from "astro:content";
...
+  schema: z.object({ // Define schema for the data. This will ensure our added terms adhere to a strict schema
+    title: z.string().min(1),
+    subtext: z.string().min(1),
+    categories: z.array(z.string()).nonempty(),
+    author: z.string().min(1),
+    description: z.object({
+      title: z.string().min(1),
+      texts: z.array(z.string()).nonempty(),
+      image: z.string().optional(),
+      references: z.array(z.string()).nonempty()
```
Comment (ManorSailor) [inline src/content/config.ts:14]:
```suggestion
      image: z.url().optional(),
      references: z.array(z.url()).nonempty()
```

We should utilize the `url` validator for concreteness

**Review by ManorSailor — CHANGES_REQUESTED** [review]

File: src/content/config.ts (hunk 1 above)
Comment (RayMathew) [reply src/content/config.ts]: Thanks @ManorSailor, it's a good suggestion 👍🏻

`z.url()` isn't valid, it's `z.string().url()`. I used this for references. 
For images I also had to combine it with being optional, which I took help from [here](https://github.com/colinhacks/zod).
`z.union([z.string().url().nullish(), z.literal("")])`

File: src/content/config.ts [hunk 2]
```diff
@@ -10,8 +10,8 @@ const termsCollection = defineCollection({
     description: z.object({
       title: z.string().min(1),
       texts: z.array(z.string()).nonempty(),
-      image: z.string().optional(),
-      references: z.array(z.string()).nonempty()
+      image: z.union([z.string().url().nullish(), z.literal("")]),
```
Comment (ManorSailor) [inline src/content/config.ts:13]: > `z.union([z.string().url().nullish(), z.literal("")])`

Do we need `nullish` here? We are already using a union which means an image can either be a `UrlString` or `""` empty literal. I just did a test [here](https://zod-playground.vercel.app?appdata=N4IgzgxgFgpgtgQxALhALwHQHsBGArGCAFwApgAdAOwAJqB3ASyKgDkBXAGw4bCmWswIATkIQBPEpjaUGWSiQDamMESENKAcxIBKDGyEcdGSp268dAGgEZuRGKMPkQT7QF1t2i1VqU57Ljx81sKiElIycorKqupauvqGnta29giOziBuHl6UAL7aIBYgAG5pbDBgKAogFELetNROjMz%2BZlBO-AomXFbSACYwAGbqMH1WTk7jIFBERAAOYMgA9EtoWH0AtHMc4hpCWP0YxfYQMBwYCHNzS06uXnU0DU6%2BlK2BHdQKE4WN07MLy1W6y2OzEewOlD6RxOZwuVxuIFc5AeuURRWOQjAskoKBAAGYMAAmAkADhAuSAA), and my assumption is correct.

On a separate note, should we really allow empty non-nullish properties - like image? It doesn't make sense for a property to exist in the first place if it has to be empty.

**Pull request merged**